src/core/common/timer.hpp

```diff
@@ -62,56 +64,70 @@ class Timer;
  */
 
 /**
- * This class implements the timer scheduler.
+ * The Alarm APIs definition
  *
  */
-class TimerScheduler
+struct AlarmApi
 {
-    friend class Timer;
+    void (*AlarmStartAt)(otInstance *aInstance, uint32_t aT0, uint32_t aDt);
+    void (*AlarmStop)(otInstance *aInstance);
+    uint32_t (*AlarmGetNow)(void);
+};
+
+/**
+ * This class implements the base timer scheduler.
+ *
+ */
+class TimerSchedulerBase: public Ip6Locator
+{
+    friend class TimerBase;
 
 public:
```

Comment:
How about if we make the constructors (and actually may be all the methods) in `TimerSchedulerBase` as `protected`? So only sub-classes can see/access them and  not allow instantiation of `TimerSchedulerBase` type object on its own.